packages/zksync-dai-bridge/contracts/l1/L1DAITokenBridge.sol

```diff
@@ -0,0 +1,216 @@
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.15;
+
+import "@matterlabs/zksync-contracts/l1/contracts/zksync/interfaces/IMailbox.sol";
+import "@matterlabs/zksync-contracts/l1/contracts/bridge/interfaces/IL1Bridge.sol";
+
+interface TokenLike {
+  function transferFrom(
+    address _from,
+    address _to,
+    uint256 _value
+  ) external returns (bool success);
+}
+
+interface L2DAITokenBridgeLike {
+  function finalizeDeposit(
+    address _from,
+    address _to,
+    address _l1Token,
+    uint256 _amount,
+    bytes calldata _data
+  ) external;
+}
+
+// Managed locked funds in L1Escrow and send / receive messages to L2DAITokenBridge counterpart
+// Note: when bridge is closed it will still process in progress messages
+
+contract L1DAITokenBridge is IL1Bridge {
+  // --- Auth ---
+  mapping(address => uint256) public wards;
+
+  function rely(address usr) external auth {
+    wards[usr] = 1;
+    emit Rely(usr);
+  }
+
+  function deny(address usr) external auth {
+    wards[usr] = 0;
+    emit Deny(usr);
+  }
+
+  modifier auth() {
+    require(wards[msg.sender] == 1, "L1DAITokenBridge/not-authorized");
+    _;
+  }
+
+  uint256 constant DEPOSIT_ERGS_LIMIT = 2097152;
+  address constant BOOTLOADER_ADDRESS = 0x0000000000000000000000000000000000008001; // address(SYSTEM_CONTRACTS_OFFSET + 0x01);
+  address public immutable l1Token;
+  address public immutable l2Bridge;
+  address public immutable l2Token;
+  address public immutable escrow;
+  IMailbox public immutable zkSyncMailbox;
+  uint256 public isOpen = 1;
+
+  mapping(uint256 => mapping(uint256 => bool)) public isWithdrawalFinalized;
+  mapping(address => mapping(bytes32 => uint256)) depositAmount;
+
+  event Rely(address indexed usr);
+  event Deny(address indexed usr);
+  event Closed();
+
+  constructor(
+    address _l1Token,
+    address _l2DAITokenBridge,
+    address _l2Token,
+    address _escrow,
+    IMailbox _mailbox
+  ) {
+    wards[msg.sender] = 1;
+    emit Rely(msg.sender);
+
+    l1Token = _l1Token;
+    l2Bridge = _l2DAITokenBridge;
+    l2Token = _l2Token;
+    escrow = _escrow;
+    zkSyncMailbox = _mailbox;
+  }
+
+  function l2TokenAddress(address _l1Token) public view returns (address) {
+    require(_l1Token == l1Token, "L1DAITokenBridge/token-not-dai");
+    return l2Token;
+  }
```

Comment:
What's up with this? Is it for compatibility? Why is it marked as `public` and not `external`?

Reply:
It's required by IBridge. Changed to `external` as it's not used by our codebase.

Reply:
Ah I see.